client/src/App.js

```diff
@@ -1,19 +1,27 @@
 import React from 'react'
-import { Route, Switch } from 'react-router-dom'
+import { Redirect, Route, Switch } from 'react-router-dom'
 import { AuthContextProvider } from './auth/AuthContext'
 import AuthorizedRoute from './auth/AuthorizedRoute'
 import GitHubLogin from './auth/GitHubLogin'
 import GitHubRedirect from './auth/GitHubRedirect'
-import Dashboard from './dashboard/Dashboard'
+import { ReposContextProvider } from './contexts/ReposContext'
+import Dashboard from './pages/dashboard/Dashboard'
+import Repos from './pages/repos/Repos'
 
 export default function App() {
   return (
     <AuthContextProvider>
-      <Switch>
-        <Route path="/login" component={GitHubLogin} />
-        <Route path="/oauth/github/redirect" component={GitHubRedirect} />
-        <AuthorizedRoute path="/" component={Dashboard} />
-      </Switch>
+      <ReposContextProvider>
+        <Switch>
+          <Route path="/login" component={GitHubLogin} />
+          <Route path="/oauth/github/redirect" component={GitHubRedirect} />
+          <AuthorizedRoute path="/repo-list" component={Repos} />
```

Comment:
Very small concern, feel free to ignore.

Instead of using two different naming conventions (`repo-list` and `repos`), I would only use one. 
For routes, it is common practice to use plural `repos`. Take GitHub for example (https://github.com/lmachens?tab=repositories) 
In addition, I would not use abbreviated form in routes for better readability.

```suggestion
          <AuthorizedRoute path="/repositories" component={Repositories} />
```